packages/viewer/src/interactions/interactionApi.ts

```diff
@@ -280,4 +330,28 @@ export class InteractionApi {
   private isCoarseInputDevice(isTouch?: boolean): boolean {
     return isTouch || window.matchMedia('(pointer: coarse)').matches;
   }
+
+  private computeWorldPosition(
+    camera: Camera,
+    viewport: Dimensions.Dimensions,
+    point: Point.Point
+  ): Vector3.Vector3 {
+    const normalizedDeviceCoordinate = Vector3.create(
+      (point.x * 2.0) / viewport.width - 1,
+      1 - (point.y * 2.0) / viewport.height,
+      this.getDepth(point)
+    );
+
+    const inverseProjPoint = Matrix4.multiplyVector3(
+      camera.inverseProjectionMatrix(),
+      normalizedDeviceCoordinate
+    );
+
+    const scaledProjPoint = Vector3.scale(
+      1.0 / inverseProjPoint.w,
+      inverseProjPoint
+    );
+
+    return Matrix4.multiplyVector3(camera.inverseViewMatrix(), scaledProjPoint);
+  }
```

Comment:
Conversion of screen click to world position. This calculation follows examples from http://bookofhook.com/mousepick.pdf and loosely from https://www.khronos.org/opengl/wiki/GluProject_and_gluUnProject_code (this uses the combined view/projection matrix, and results in the same computed point).

Reply:
One additional note, a couple of these values are adjusted based on testing -- the `y` component of the normalized device coordinate is flipped to go [-1, 1] from [bottom, top]. Additionally, the `z` component of the normalized device coordinate is in the range [0, 1] instead of [-1, 1] based on experimenting with the range, since it resulted in values that were closer to the expected world position